### PR TITLE
Ignore Early Hints (4.9.x)

### PR DESCRIPTION
**Pull request by yschimke**
(cherry picked from commit b565eecf167918253145bce2c348b0bb0ef58b0c)

### CODE REVIEW

**Comment by yschimke** [comment]
@swankjesse CallTest can't be reused because of addInformationResponse which was added in 5

File: okhttp/src/main/kotlin/okhttp3/internal/http/CallServerInterceptor.kt [hunk 1]
```diff
@@ -148,4 +149,15 @@ class CallServerInterceptor(private val forWebSocket: Boolean) : Interceptor {
       throw e
     }
   }
+
+  private fun shouldIgnoreAndWaitForRealResponse(code: Int, exchange: Exchange): Boolean = when {
```
Comment (reschke) [inline okhttp/src/main/kotlin/okhttp3/internal/http/CallServerInterceptor.kt:153]: That seems to do the wrong thing for all 1xx codes except 100 and 103.

**Review by reschke — CHANGES_REQUESTED** [review]

**Comment by yschimke** [comment]
The intent was to get something in quick to fix an obviously surfaced bug affecting cloudflare customers.

Then a bigger fix in https://github.com/square/okhttp/issues/2257

It's in the blocker list for 5, https://github.com/square/okhttp/issues/7366

Is there a simple fix here we can make to expand it to all other codes including future 1xx and codes we aren't expecting?

**Comment by reschke** [comment]
Just treat all 1xx the same, except for the one that is somewhat special (101).

**Comment by yschimke** [comment]
I'll close this, make a minor update for 5.x/main.  Then backport.

**Pull request closed without merging**